src/main/java/hw25/jdbc/LessonDao.java

```diff
@@ -0,0 +1,132 @@
+            throw new JdbcOperationException("Failed to insert the lesson!", e);
+        }
+    }
+
+    public String delete(final Long id) {
+        Objects.requireNonNull(id);
+        String sql = """
+                WITH deleted_homework AS (DELETE FROM lesson WHERE id = ?
+                  RETURNING homework_id)
+                DELETE FROM homework WHERE id IN (SELECT homework_id FROM deleted_homework)
+                 AND NOT EXISTS (SELECT 1 FROM lesson WHERE homework_id = 
+                  (SELECT homework_id FROM deleted_homework) AND id <> ?)
+                """;
+        try (PreparedStatement prepStatement = DataBaseConnection.getConnection()
+                .prepareStatement(sql)) {
+            int index = 1;
+            prepStatement.setLong(index++, id);
+            prepStatement.setLong(index, id);
+            int result = prepStatement.executeUpdate();
+            if (result == 1) {
+                return "Lesson and related homework were successfully deleted";
+            } else if (result == 0) {
+                return "Lesson was successfully deleted. Homework wasn't deleted because " +
+                        "other lesson has this homework also.";
+            } else {
+                throw new JdbcOperationException("No rows were deleted!");
+            }
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Failed to delete a lesson by ID: " + id, e);
+        }
+    }
+
+
+    public List<Lesson> getAll() {
+        List<Lesson> lessons = new ArrayList<>();
+        String sql = """
+                SELECT l.id, l.name, l.updated_at, h.id, h.name, h.description
+                FROM lesson l LEFT JOIN homework h ON l.homework_id = h.id;
+                """;
+        try (PreparedStatement prepStatement = DataBaseConnection.getConnection()
+                .prepareStatement(sql)) {
+            ResultSet resultSet = prepStatement.executeQuery();
+            while (resultSet.next()) {
+                int index = 1;
+                Lesson lesson = retreiveLesson(resultSet, index);
+                lessons.add(lesson);
+            }
+            return lessons;
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Failed to get all lessons!", e);
+        }
+    }
+
+    public Lesson getOne(final Long id) {
+        Lesson lesson = new Lesson();
+        String sql = """
+                SELECT l.id, l.name, l.updated_at, h.id, h.name, h.description
+                FROM lesson l LEFT JOIN homework h ON l.homework_id = h.id
+                WHERE l.id = ?
+                """;
+        try (PreparedStatement prepStatement = DataBaseConnection.getConnection()
+                .prepareStatement(sql)) {
+            int setIndex = 1;
+            prepStatement.setLong(setIndex, id);
+            ResultSet resultSet = prepStatement.executeQuery();
+            if (resultSet.next()) {
+                int getIndex = 1;
```

Comment:
As was mentioned you can move this `getIndex` under the scope of `retrieveLesson` method, so you can get rid of the weird variable names.